exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/KeyValueMarshaler.java

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.otlp.internal;
+
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.proto.common.v1.internal.AnyValue;
+import io.opentelemetry.proto.common.v1.internal.ArrayValue;
+import io.opentelemetry.proto.common.v1.internal.KeyValue;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.function.BiConsumer;
+
+final class KeyValueMarshaler extends MarshalerWithSize {
+  private static final KeyValueMarshaler[] EMPTY_REPEATED = new KeyValueMarshaler[0];
+
+  static KeyValueMarshaler[] createRepeated(Attributes attributes) {
+    if (attributes.isEmpty()) {
+      return EMPTY_REPEATED;
+    }
+
+    KeyValueMarshaler[] attributeMarshalers = new KeyValueMarshaler[attributes.size()];
+    attributes.forEach(
+        new BiConsumer<AttributeKey<?>, Object>() {
+          int index = 0;
+
+          @Override
+          public void accept(AttributeKey<?> attributeKey, Object o) {
+            attributeMarshalers[index++] = KeyValueMarshaler.create(attributeKey, o);
+          }
+        });
+    return attributeMarshalers;
+  }
+
+  private final byte[] keyUtf8;
+  private final Marshaler value;
+
+  @SuppressWarnings("unchecked")
+  static KeyValueMarshaler create(AttributeKey<?> attributeKey, Object value) {
+    byte[] keyUtf8 = MarshalerUtil.toBytes(attributeKey.getKey());
+    switch (attributeKey.getType()) {
+      case STRING:
+        return new KeyValueMarshaler(
+            keyUtf8, new AnyStringFieldMarshaler(MarshalerUtil.toBytes((String) value)));
+      case LONG:
+        return new KeyValueMarshaler(keyUtf8, new AnyInt64FieldMarshaler((long) value));
```

Comment:
Note the extra `new` compared to previous, one extra wrapper because of the oneof. In the future this could be removed using another switch statement instead elsewhere if it's worth it